### PR TITLE
Add `checkFilenames` option to the `prevent-abbreviations` rule

### PR DESCRIPTION
**Pull request by fisker**
fixes: #267

### CODE REVIEW

File: rules/prevent-abbreviations.js [hunk 1]
```diff
@@ -178,6 +180,16 @@ const defaultWhitelist = {
 	stdDev: true
 };
 
+const getCase = string => {
+	for (const fn of [camelCase, kebabCase, snakeCase, upperfirst]) {
```
Comment (futpib) [inline rules/prevent-abbreviations.js]: Shouldn't this use `pascalCase` instead of `upperfirst`?

File: test/prevent-abbreviations.js [hunk 2]
```diff
@@ -911,6 +926,17 @@ ruleTester.run('prevent-abbreviations', rule, {
...
+		},
+		// `checkFilenames` option
+		{
+			code: 'foo();',
+			filename: 'err/http-err.js',
+			errors: createErrors()
+		},
+		{
+			code: 'foo();',
+			filename: 'http-err.js',
```
Comment (futpib) [inline test/prevent-abbreviations.js]: Could we also add a test for each kind of case from `getCase`? And also a test with an underscore (`_helper.js`) and dot (`.babelrc.js`) prefixes and a dot separator (`next.config.js`)

File: rules/prevent-abbreviations.js (hunk 1 above)
Comment (fisker) [reply rules/prevent-abbreviations.js]: will send another PR to fix it

File: test/prevent-abbreviations.js (hunk 2 above)
Comment (fisker) [reply test/prevent-abbreviations.js]: will add more tests

**Pull request merged**